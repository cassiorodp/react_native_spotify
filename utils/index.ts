import * as SecureStore from 'expo-secure-store';

const secureStore = async (key: string, value: string) => {
  await SecureStore.setItemAsync(key, value);
};

const getSecureStore = (key: string) => {
  return SecureStore.getItem(key);
};

export { secureStore, getSecureStore };
