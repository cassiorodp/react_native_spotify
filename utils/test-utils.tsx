import {
  RenderAPI,
  RenderOptions,
  render,
} from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from 'react-query';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  options?: RenderOptions
): RenderAPI => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
