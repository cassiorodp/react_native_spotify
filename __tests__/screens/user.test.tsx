import { fireEvent, waitFor } from '@testing-library/react-native';
import { clearSecureStore } from '@/utils';
import { router } from 'expo-router';
import { render } from '@/utils/test-utils';
import User from '@/app/(tabs)/user';

jest.mock('expo-router', () => ({
  router: {
    replace: jest.fn(),
  },
}));

jest.mock('@/utils', () => ({
  clearSecureStore: jest.fn(),
}));

jest.mock('@/api/endpoints', () => ({
  getUserData: jest.fn().mockResolvedValue({
    data: {
      display_name: 'Test User',
      images: [{ url: 'https://example.com/image.jpg' }],
    },
  }),
}));

beforeEach(() => {
  jest.clearAllMocks();
});

describe('User screen', () => {
  it('renders the matching snapshot', () => {
    const { toJSON } = render(<User />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders user data and logs out on button press', async () => {
    const { getByText, findByText } = render(<User />);

    expect(await findByText('Test User')).toBeTruthy();

    fireEvent.press(getByText('Sair'));

    expect(clearSecureStore).toHaveBeenCalled();
    await waitFor(() => expect(router.replace).toHaveBeenCalledWith('/'));
  });
});
