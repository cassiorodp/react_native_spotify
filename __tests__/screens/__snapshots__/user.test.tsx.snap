// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User screen renders the matching snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "gap": 16,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="rgb(0, 122, 255)"
      size="large"
    />
    <Text
      style={
        [
          {
            "color": "rgb(28, 28, 30)",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontFamily": "Rubik-Medium",
            "fontSize": 24,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "rgb(0, 122, 255)",
        "borderRadius": 100,
        "marginTop": 10,
        "opacity": 1,
        "paddingHorizontal": 40,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "rgb(28, 28, 30)",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          [
            {
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
            },
            {
              "color": "rgb(242, 242, 242)",
            },
          ],
          undefined,
          undefined,
        ]
      }
    >
      Sair
    </Text>
  </View>
</RNCSafeAreaView>
`;
